### ISSUE TITLE
Missing aarch64 wheels for python 3.10

### ISSUE THREAD

**Issue by thomasaarholt** [Unidata/netcdf4-python#1169]
Trying to install netcdf4 1.5.8 in docker running on a M1 Mac. On 3.9 everything works smoothly, on 3.10 it fails with the error below.
On [pypi](https://pypi.org/project/netCDF4/#files) I see that aarch64 wheels for python 3.10 are missing.

```python
# Error message
Collecting netcdf4
  Using cached netCDF4-1.5.8.tar.gz (767 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [23 lines of output]
      reading from setup.cfg...
      
          HDF5_DIR environment variable not set, checking some standard locations ..
      checking /root/include ...
      hdf5 headers not found in /root/include
      checking /usr/local/include ...
      hdf5 headers not found in /usr/local/include
      checking /sw/include ...
      hdf5 headers not found in /sw/include
      checking /opt/include ...
      hdf5 headers not found in /opt/include
      checking /opt/local/include ...
      hdf5 headers not found in /opt/local/include
      checking /usr/include ...
      hdf5 headers not found in /usr/include
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-g6yd58tq/netcdf4_976f18b55de742c59d056d6427c1b438/setup.py", line 419, in <module>
          _populate_hdf5_info(dirstosearch, inc_dirs, libs, lib_dirs)
        File "/tmp/pip-install-g6yd58tq/netcdf4_976f18b55de742c59d056d6427c1b438/setup.py", line 360, in _populate_hdf5_info
          raise ValueError('did not find HDF5 headers')
      ValueError: did not find HDF5 headers
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

If you're using `pip-compile` to generate a `requirements.txt` with netcdf4, you get `pip._internal.exceptions.MetadataGenerationFailed: metadata generation failed`, which is an awful error message that I am including here in case anyone wishes to learn that this error message means that one of your packages doesn't want to install, and you are best off checking compatibility for each package.

**Comment by jswhit2** [comment]
This is actually a bug in https://github.com/MacPython/netcdf4-python-wheels - the aarch64 build for python 3.10 failed (https://github.com/MacPython/netcdf4-python-wheels/runs/5637004673?check_suite_focus=true).  This will be fixed in the next relaase (1.6.0, which will happen as soon as netcdf-c 4.9.0 is released)

**Comment by jswhit** [comment]
1.6.0 has been released with aarch64 wheels for python 3.10
